### PR TITLE
Added vargs into waitUntil done

### PR DESCRIPTION
**Pull request by nikita-leonov**
Currently done does not accept any arguments, as result sometimes it require additional boilerplate code. The current pull request solves this problem.

Here is an example what we can do with current waitUntil implementation:
```
func asyncCall(callback:(Void -> Void)) { ... }
func argsAsyncCall(callback:(AnyObject->Void)) { ... }

waitUntil { done in
  asyncCall(done) //Looks nice
  asyncCall { _ in done() } //Looks ugly
}
```

Proposed change allows to write code consistently since introduces variadic arguments for done function. Swift allows zero to any amount of variadic arguments. As result we always getting consistent code: 
```
waitUntil { done in 
  anyAsyncCall(done) // consistency!
}
```

### CODE REVIEW

**Comment by modocache** [comment]
@nikita-leonov `master` is indeed the branch you want! `swift-2.0` was merged into `master` a week ago, in #187. I'll reopen your request against master, #192.

**Comment by modocache** [comment]
Oops, looks like I can't...

screen shot 2015-09-23 at 11 38 17 pm

File: Nimble/DSL+Wait.swift [hunk 1]
```diff
@@ -4,7 +4,7 @@ import Foundation
 /// bridges to Objective-C via the @objc keyword. This class encapsulates callback-style
 /// asynchronous waiting logic so that it may be called from Objective-C and Swift.
 internal class NMBWait: NSObject {
-    internal class func until(timeout timeout: NSTimeInterval, file: String = __FILE__, line: UInt = __LINE__, action: (() -> Void) -> Void) -> Void {
+    internal class func until(timeout timeout: NSTimeInterval, file: String = __FILE__, line: UInt = __LINE__, action: (done:() -> Void) -> Void) -> Void {
```
Comment (modocache) [inline Nimble/DSL+Wait.swift]: nit: Add a space in between `done:` and `()`, as in `done: () -> Void`.

File: Nimble/DSL+Wait.swift [hunk 2]
```diff
@@ -28,7 +29,7 @@ internal class NMBWait: NSObject {
             break
         }
     }
-
+
```
Comment (modocache) [inline Nimble/DSL+Wait.swift]: nit: Remove this extra whitespace.

File: Nimble/DSL+Wait.swift [hunk 3]
```diff
@@ -38,13 +39,22 @@ internal class NMBWait: NSObject {
...
 /// Wait asynchronously until the done closure is called.
 ///
 /// This will advance the run loop.
-public func waitUntil(file: String = __FILE__, line: UInt = __LINE__, action: (() -> Void) -> Void) -> Void {
-    NMBWait.until(timeout: 1, file: file, line: line, action: action)
+public func waitUntil(file: String = __FILE__, line: UInt = __LINE__, action: ((Any...) -> Void) -> Void) -> Void {
+    NMBWait.until(timeout: 1, file: file, line: line, action: bridgeToVoidAction(action))
+}
+
+//This bridging function should not be required in Swift 2.1 as its support functions covariance
```
Comment (modocache) [inline Nimble/DSL+Wait.swift]: nit: Typo, should be "as it supports function covariance".

**Comment by modocache** [comment]
Could you include a unit test in your PR that demonstrates the new, consistent syntax? That way we can ensure that this change doesn't regress in future releases.

**Comment by modocache** [comment]
Overall this seems like a reasonable change to me--I just had a few suggestions and nitpicks. I defer to @jeffh, in case he has any thoughts on whether this should be included.

Sorry again for closing this pull request! Please re-submit it against the `master` branch if you don't mind. Otherwise, once it's reviewed one of us can merge it manually into `master`.

**Comment by nikita-leonov** [comment]
@modocache Sure thing. I will redo it for master, as well as address your suggestions and will add unit tests. Thanks!

**Pull request closed without merging**